clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -244,11 +247,11 @@ public void updateFetchPositions(Set<TopicPartition> partitions) {
                 continue;
 
             if (subscriptions.isOffsetResetNeeded(tp)) {
-                resetOffset(tp);
+                resetOffsets(Collections.singleton(tp));
```

Comment:
Currently `updateFetchPositions` is always called after `resetOffsetsIfNeeded` so these branch should not be hit in practice other than unit tests, but I think we can still batch them if necessary or simply throw an exception if `subscriptions.hasValidPosition(tp) && subscriptions.committed(tp) == null`. WDYT @hachikuji ?